pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +93,539 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+	a.behaviour.greediness -= int(float64(a.opMem.currentFloorRisk) * (float64(a.personality.conscientiousness) / 50))
+	foodReqCtoW := int(health.FoodRequired(a.HealthInfo().HPCritical, a.HealthInfo().WeakLevel, a.HealthInfo()))
+	if expectedFood < foodReqCtoW { //greediness only begins to be effected once the expected food is below a crticial threshold
+		a.opMem.currentFloorRisk = (foodReqCtoW - expectedFood)
+		a.behaviour.greediness += int(float64(a.opMem.currentFloorRisk) * (float64(a.personality.conscientiousness) / 50))
+	} else {
+		a.opMem.currentFloorRisk = 0
+	}
+	a.treatyOnFloorChange()
+}
+
+func (a *CustomAgent7) manageMessages() {
+	// ------------------ Run() Block B: Receive messages ------------------
+	receivedMsg := a.ReceiveMessage()
+	if receivedMsg != nil {
+		receivedMsg.Visit(a)
+	} else {
+		a.Log("No Messages")
+	}
+
+	// Messaging
+	if a.opMem.prevAge < a.Age() {
+		a.opMem.prevAge = a.Age()
+		a.opMem.msg1Sent = false
+		a.opMem.msg2Sent = false
+		a.opMem.msg3Sent = false
+	}
+
+	if !a.opMem.msg1Sent {
+		msg := messages.NewAskHPMessage(a.ID(), a.Floor(), a.Floor()+1)
+		// a.opMem.messageQueue = append(a.opMem.messageQueue, msg)
+		a.SendMessage(msg)
+		if a.Floor() != 1 {
+			msg = messages.NewAskHPMessage(a.ID(), a.Floor(), a.Floor()-1)
+			a.SendMessage(msg)
 		}
-		a.prevFloors = append(a.prevFloors, currentFloor)
+		a.opMem.msg1Sent = true
 	}
 
-	//receive Message
+	if a.PlatformOnFloor() && !a.opMem.msg2Sent {
+		if a.Floor() != 1 {
+			msg2 := messages.NewAskFoodTakenMessage(a.ID(), a.Floor(), a.Floor()-1)
+			a.SendMessage(msg2)
+		}
+		a.opMem.msg2Sent = true
+	}
+
+	if !a.PlatformOnFloor() && a.CurrPlatFood() != -1 && a.opMem.seenPlatform && a.Floor() != 1 && !a.opMem.msg3Sent {
+		msg2 := messages.NewAskFoodTakenMessage(a.ID(), a.Floor(), a.Floor()+1)
+		a.SendMessage(msg2)
+		a.opMem.msg3Sent = true
+	}
+
+}
+
+func (a *CustomAgent7) manageNewDay() {
 
-	//eat
+	a.opMem.seenPlatform = false
+	a.opMem.prevHP = a.HP()
+	a.opMem.foodEaten = 0
+	a.opMem.receivedReq = false
 
-	//choses random number between -5 and 5
-	if a.CurrPlatFood() != -1 {
+}
 
-		r1 := rand.Intn(11) - 5
-		r2 := rand.Intn(11) - 5
+func (a *CustomAgent7) treatyGT(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
+
+	if a.HP() > tActive.ConditionValue() || int(a.CurrPlatFood()) > tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		}
```

Comment:
You've copied this same block of code five times - you can put this into a function.

Reply:
You can look at this to see something similar: https://github.com/SOMAS2021/SOMAS2021/pull/289/files#diff-2124353f8301e3dd24472c40d43b634e52f647d7538af55a0770d95b97e3ef48R224-R239